packages/asset-kit/src/AssetKitBlock.tsx

```diff
@@ -0,0 +1,81 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { joinClassNames, useGuidelineDesignTokens } from '@frontify/guideline-blocks-shared';
+import '@frontify/fondue-tokens/styles';
+import { BlockProps } from '@frontify/guideline-blocks-settings';
+import { ReactElement, useState } from 'react';
+import 'tailwindcss/tailwind.css';
+import { useBlockAssets, useBlockSettings, useEditorState } from '@frontify/app-bridge';
+import { ASSET_SETTINGS_ID } from './settings';
+import { Settings } from './types';
+import { AssetGrid, AssetSelection, DownloadMessage, InformationSection } from './components';
+import { blockStyle, generateBulkDownload, thumbnailStyle } from './helpers';
+
+export const AssetKitBlock = ({ appBridge }: BlockProps): ReactElement => {
+    const { designTokens } = useGuidelineDesignTokens();
+    const [blockSettings, setBlockSettings] = useBlockSettings<Settings>(appBridge);
+    const isEditing = useEditorState(appBridge);
+    const { blockAssets, addAssetIdsToKey, deleteAssetIdsFromKey, updateAssetIdsFromKey } = useBlockAssets(appBridge);
+    const [isUploadingAssets, setIsUploadingAssets] = useState<boolean>(false);
+    const [isDownloadingAssets, setIsDownloadingAssets] = useState<boolean>(false);
+    const { title, description } = blockSettings;
+    const currentAssets = blockAssets[ASSET_SETTINGS_ID] ?? [];
+    const { hasBorder_blocks, hasBackground_blocks } = blockSettings;
+
+    return (
+        <div
+            data-test-id="asset-kit-block"
+            className={joinClassNames(['tw-w-full', (hasBorder_blocks || hasBackground_blocks) && 'tw-p-5 sm:tw-p-8'])}
+            style={{
+                ...blockStyle(blockSettings),
+            }}
+        >
+            <div className="tw-mb-8 sm:tw-flex tw-gap-8 tw-space-y-3 md:tw-space-y-0">
+                <InformationSection
+                    description={description ?? ''}
+                    isEditing={isEditing}
+                    setBlockSettings={setBlockSettings}
+                    title={title ?? ''}
+                />
+                <div className="tw-flex-none">
+                    <button
+                        data-test-id="asset-kit-block-download-button"
+                        disabled={
+                            isDownloadingAssets ||
+                            isUploadingAssets ||
+                            currentAssets.length === undefined ||
+                            currentAssets.length <= 0
+                        }
```

Comment:
No need to check for `undefined` as it's declared on line 22.
```suggestion
                        disabled={
                            isDownloadingAssets ||
                            isUploadingAssets ||
                            currentAssets.length <= 0
                        }
```

Reply:
adjusted